### PR TITLE
Add debug-level Zebra network message tracing

### PR DESCRIPTION
**Pull request by teor2345**
## Motivation

As part of #3136, #2726, and #2156, we need to trace Zebra network messages to and from peers.

This PR does not close any tickets.

## Solution

Add debug-level tracing for:
- Zebra requests to peers, and peer responses
- Peer requests to Zebra, and Zebra responses

## Review

Anyone can review this low-priority PR, but maybe @oxarbitrage has some time to review it?

### Reviewer Checklist

  - [ ] Code produces useful diagnostics

## Follow Up Work

- #3136
- #2726 
- #2156

### CODE REVIEW

File: zebra-network/src/peer/connection.rs [hunk 1]
```diff
@@ -533,6 +539,13 @@ where
                                     tx,
                                     ..
                                 } => {
+
```
Comment (oxarbitrage) [inline zebra-network/src/peer/connection.rs]:
```suggestion
```

File: zebra-network/src/peer/connection.rs [hunk 2]
```diff
@@ -884,6 +899,9 @@ where
     #[instrument(name = "msg_as_req", skip(self, msg), fields(%msg))]
     async fn handle_message_as_request(&mut self, msg: Message) {
         trace!(?msg);
+
```
Comment (oxarbitrage) [inline zebra-network/src/peer/connection.rs]:
```suggestion
```

Is the `trace!` and the `debug!` needed ? If so, maybe remove the space between them ?

**Review by oxarbitrage — DISMISSED** [review]
Looks good, just a few minor spacing issues but they are not important, feel free to discard them.

File: zebra-network/src/peer/connection.rs (hunk 2 above)
Comment (teor2345) [reply zebra-network/src/peer/connection.rs]: Yes, `%msg` is a shorter summary of the message, `?msg` is the full message data.

**Review by oxarbitrage — APPROVED** [review]

**Pull request merged**